syft/pkg/cataloger/apkdb/parse_apk_db.go

```diff
@@ -190,20 +190,20 @@ func discoverPackageDependencies(pkgs []*pkg.Package) (relationships []artifact.
 	for _, p := range pkgs {
 		apkg, ok := p.Metadata.(pkg.ApkMetadata)
 		if !ok {
-			// TODO warn
+			log.Warnf("cataloger failed to extract apk 'provides' metadata for package %+v", p.Name)
 			continue
 		}
 		for _, provides := range apkg.Provides {
-			// TODO: parse provides to remove =*
-			lookup[provides] = append(lookup[provides], p)
+			k := strings.Split(provides, "=")[0]
```

Comment:
This is how I'd do it in Python, not sure if there is a more idiomatic way in Go.

Reply:
the `=` is correct (as it is in the code now). Take for example: 

```
P:ssl_client
V:1.35.0-r17
D:so:libc.musl-x86_64.so.1 so:libcrypto.so.1.1 so:libssl.so.1.1.       # provides libcrypto
```
and
```
P:libcrypto1.1
V:1.1.1q-r0
D:so:libc.musl-x86_64.so.1
p:so:libcrypto.so.1.1=1.1           # depends on libcrypto v1.1 specifically
```

Where `ssl_client` depends on `libcrypto`. The libcrypto package specifies the exact version provided and the consumer specifies the loose requirements that would suffice (in this case there is no `=1.1`). A better example is one where the provided specifies a more specific version, and the consumer locks to the major version (and not the minor version).

But, let's take a look at another example:
```
C:Q1VFFFWMKjB9aRkehIATc5kwgAhlU=
P:apk-tools
V:2.12.9-r3
A:x86_64
D:musl>=1.2 ca-certificates-bundle so:libc.musl-x86_64.so.1 so:libcrypto.so.1.1 so:libssl.so.1.1 so:libz.so.1
p:so:libapk.so.3.12.0=3.12.0 cmd:apk=2.12.9-r3
```
where `musl` is referring to a package name (`P:` field) and not a provider field (`p:` field):
```
C:Q1aCu0LmUDoAFSOX49uHvkYC1WasQ=
P:musl
V:1.2.3-r0
# note, there is no D: or p: field on this entry
```

We want the "dependencies" and "provides" to match up, like with the previous example, but this shows that we need "dependencies" and the "package" field to also be sometimes related.

This hints at more things needing to be captured by this parser.